lib/foreman_tasks_core/task_launcher/batch.rb

```diff
@@ -27,11 +27,31 @@ def launch!(input)
 
       def launch_children(parent, input_hash)
         input_hash.each do |task_id, input|
-          launcher = Single.new(world, callback, :parent => parent)
-          launcher.launch!(input)
+          launcher = child_launcher(parent)
+          launcher.launch!(transform_input(input))
           results[task_id] = launcher.results
         end
       end
+
+      def prepare_batch(input_hash)
```

Comment:
Is there some better name for this method: took me a while to figure out what it's supposed to do.

I think it would help the the launched not having to know about the callbacks, and just provide a method `planned_task_ids` that would return the task ids that were successfully planned, and use it in `GroupParentAction` to prepare the callback input